### PR TITLE
api docs: document stream format

### LINKED ISSUE #19280: Why am I receiving random bytes over podman.socket
### Issue Description

I sometimes receive a sequence like this: 
`\SOH\NUL\NUL\NUL\NUL\NUL\NULb`
which breaks the json that my script in the exec is outputting. 
My first workaround was to throw away the first 8 bytes but this then reoccurs later in the stream.

### Steps to reproduce the issue

Steps to reproduce the issue

1. send
```
POST /containers/{cid}/exec HTTP/1.1
Host: localhost
User-Agent: something
Content-Type: application/json
Content-Length: 196

{"AttachStderr":true,"AttachStdin":false,"AttachStdout":true,"Cmd":["/bin/bash","-c","ls"],"DetachKeys":"ctrl-a","Logs":false,"Stream":true,"Tty":false,"User":"root","WorkingDir":"/tmp/compiling"}
```
2. send
```
POST /exec/{responseIdFromStep1}/start HTTP/1.1
Host: localhost
User-Agent: esystant
Content-Type: application/json
Content-Length: 27

{"Detach":false,"Tty":true}
```

3. Observe weird bytes prepended. 
4. I've also noticed that after reading a lot of bytes, this re-occurs.

### Describe the results you received

This is what I get:

[file1.txt](https://github.com/containers/podman/files/12089248/file1.txt)

### Describe the results you expected

This is what I expect to get:

[file1.txt](https://github.com/containers/podman/files/12089250/file1.txt)

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.30.0
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: /usr/bin/conmon is owned by conmon 1:2.1.7-1
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: f633919178f6c8ee4fb41b848a056ec33f8d707d'
  cpuUtilization:
    idlePercent: 97.19
    systemPercent: 0.4
    userPercent: 2.41
  cpus: 12
  databaseBackend: boltdb
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  hostname: arch
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 6.4.3-arch1-2
  linkmode: dynamic
  logDriver: journald
  memFree: 10285826048
  memTotal: 33556062208
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: /usr/bin/crun is owned by crun 1.8.5-1
    path: /usr/bin/crun
    version: |-
      crun version 1.8.5
      commit: b6f80f766c9a89eb7b1440c0a70ab287434b17ed
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: /usr/bin/slirp4netns is owned by slirp4netns 1.2.0-1
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.4
  swapFree: 8589668352
  swapTotal: 8589930496
  uptime: 11h 28m 53.00s (Approximately 0.46 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/merlijn/.config/containers/storage.conf
  containerStore:
    number: 5
    paused: 0
    running: 2
    stopped: 3
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/merlijn/.local/share/containers/storage
  graphRootAllocated: 973837332480
  graphRootUsed: 832484134912
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 33
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/merlijn/.local/share/containers/storage/volumes
version:
  APIVersion: 4.5.1
  Built: 1685139594
  BuiltTime: Sat May 27 00:19:54 2023
  GitCommit: 9eef30051c83f62816a1772a743e5f1271b196d7-dirty
  GoVersion: go1.20.4
  Os: linux
  OsArch: linux/amd64
  Version: 4.5.1
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

No

### Additional environment details

Arch linux

### Additional information

I used `nc -U /run/user/1000/podman/podman.sock > file1.txt 2>&1` to send the http command and create the output files

### PR DESCRIPTION
**Pull request by Luap99**
Document the attach, exec and logs output stream format. We use the same format as docker.

Fixes #19280

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/api/server/register_containers.go [hunk 1]
```diff
@@ -526,7 +526,95 @@ func (s *APIServer) registerContainersHandlers(r *mux.Router) error {
...
+	//   Connection: Upgrade
+	//   Upgrade: tcp
+	//
+	//   [STREAM]
+	//   ```
+	//
+	//   ### Stream format
+	//
+	//   When the TTY setting is disabled for the container,
+	//   the HTTP Content-Type header is set to application/vnd.docker.multiplexed-stream
```
Comment (vrothberg) [inline pkg/api/server/register_containers.go:577]: I am unable to find track down where this is being set in Podman.  I see it in vendor.

**Review by vrothberg — COMMENTED** [review]
The redundancy in the swagger docs is unfortunate but nothing we can solve in this PR.  So LGTM but I have one comment on the media type.

File: pkg/api/server/register_containers.go (hunk 1 above)
Comment (Luap99) [reply pkg/api/server/register_containers.go:577]: Good catch, yes looks like we always set `application/vnd.docker.raw-stream` regardless of tty or not.
I guess nobody really cares about the headers, I will see if I can fix it in the code. Feels weird to me that we even use these docker headers in the libpod API but it is what it is.

Comment (Luap99) [reply pkg/api/server/register_containers.go:577]: ah looks like docker just added this type with api 1.42, https://github.com/moby/moby/blob/331854a126674175dde199c6516174d15862cf1c/api/server/router/container/container_routes.go#L160-L162

File: pkg/api/server/register_containers.go [hunk 2]
```diff
@@ -526,7 +526,95 @@ func (s *APIServer) registerContainersHandlers(r *mux.Router) error {
...
+	//   HTTP/1.1 200 OK
+	//   Content-Type: application/vnd.docker.raw-stream
+	//
+	//   [STREAM]
+	//   ```
+	//
+	//   After the headers and two new lines, the TCP connection can now be used
+	//   for raw, bidirectional communication between the client and server.
+	//
+	//   To hint potential proxies about connection hijacking, the client
```
Comment (mheon) [inline pkg/api/server/register_containers.go:551]: "To inform" instead of "To hint" maybe?

File: pkg/api/server/register_containers.go [hunk 3]
```diff
@@ -526,7 +526,95 @@ func (s *APIServer) registerContainersHandlers(r *mux.Router) error {
...
+	//
+	//   When the TTY setting is disabled for the container,
+	//   the HTTP Content-Type header is set to application/vnd.docker.multiplexed-stream
+	//   and the stream over the hijacked connected is multiplexed to separate out
+	//   `stdout` and `stderr`. The stream consists of a series of frames, each
+	//   containing a header and a payload.
+	//
+	//   The header contains the information which the stream writes (`stdout` or
+	//   `stderr`). It also contains the size of the associated frame encoded in
+	//   the last four bytes (`uint32`).
```
Comment (mheon) [inline pkg/api/server/register_containers.go:584]: Should mention byte-order of the u32 here. Not clear if we're sending little or big endian in this.

Comment (mheon) [reply pkg/api/server/register_containers.go:584]: I see that this is mentioned a few paragraphs down... Maybe unify that paragraph into here?

File: pkg/api/server/register_containers.go [hunk 4]
```diff
@@ -1319,7 +1410,95 @@ func (s *APIServer) registerContainersHandlers(r *mux.Router) error {
 	// tags:
 	//   - containers
 	// summary: Attach to a container
-	// description: Hijacks the connection to forward the container's standard streams to the client.
+	// description: |
+	//   Attach to a container to read its output or send it input. You can attach
```
Comment (mheon) [inline pkg/api/server/register_containers.go:1414]: Maybe just mention that the format is identical to Compat Attach?

File: pkg/api/server/register_containers.go (hunk 2 above)
Comment (Luap99) [reply pkg/api/server/register_containers.go:551]: Sure this was mostly copied from the docker docs, I just removed the docker daemon references.

File: pkg/api/server/register_containers.go (hunk 4 above)
Comment (Luap99) [reply pkg/api/server/register_containers.go:1414]: Properly better to do it the other way around? I mean I expect only a small number of people to looks at out compat docs. I would assume most people would use the docker docs or the libpod specific endpoint docs. SO having it here sounds better to me.

**Comment by Luap99** [comment]
PTAL again, I removed the duplicated docs and added code the return the new docker header. I decided to do for libpod endpoint as well as it will allow optimizations on the client side.

**Pull request merged**